### PR TITLE
 Split event creation into a separate handler

### PR DESCRIPTION
**Pull request by erikjohnston**

### CODE REVIEW

**Comment by erikjohnston** [comment]
@matrixbot retest this please

File: synapse/handlers/message.py [hunk 1]
```diff
@@ -342,134 +486,6 @@ def create_and_send_nonmember_event(
...
-            user_id: {
-                "avatar_url": profile.avatar_url,
-                "display_name": profile.display_name,
-            }
-            for user_id, profile in users_with_profile.iteritems()
-        })
-
     @measure_func("_create_new_client_event")
     @defer.inlineCallbacks
     def _create_new_client_event(self, builder, requester=None, prev_event_ids=None):
```
Comment (richvdh) [inline synapse/handlers/message.py]: if we're moving this anyway, and it's intended to be called from outside the handler, please can we drop the leading `_` ?

File: synapse/handlers/message.py [hunk 2]
```diff
@@ -182,6 +170,162 @@ def get_messages(self, requester, room_id=None, pagin_config=None,
...
+        self.state = hs.get_state_handler()
+        self.clock = hs.get_clock()
+        self.validator = EventValidator()
+        self.profile_handler = hs.get_profile_handler()
+        self.event_builder_factory = hs.get_event_builder_factory()
+        self.server_name = hs.hostname
+        self.ratelimiter = hs.get_ratelimiter()
+        self.notifier = hs.get_notifier()
+
+        # This is only used to get at ratelimit function, and maybe_kick_guest_users
```
Comment (richvdh) [inline synapse/handlers/message.py]: I kinda wish it wish they were factored out, but I guess that is a problem for another time

File: synapse/handlers/message.py [hunk 3]
```diff
@@ -182,6 +170,162 @@ def get_messages(self, requester, room_id=None, pagin_config=None,
...
+                "avatar_url": profile.avatar_url,
+                "display_name": profile.display_name,
+            }
+            for user_id, profile in users_with_profile.iteritems()
+        })
+
+
+class EventCreationHandler(object):
+    def __init__(self, hs):
+        self.hs = hs
```
Comment (richvdh) [inline synapse/handlers/message.py]: it'd be nice if all these private fields were named with a leading underscore, but I suppose that increases the diff significantly?

File: synapse/handlers/message.py [hunk 4]
```diff
@@ -378,7 +378,7 @@ def create_event(self, requester, event_dict, token_id=None, txn_id=None,
             if txn_id is not None:
                 builder.internal_metadata.txn_id = txn_id
 
-            event, context = yield self._create_new_client_event(
+            event, context = yield self.create_new_client_event(
```
Comment (richvdh) [inline synapse/handlers/message.py]: yeah, you see you made the mistake of expecting the thing that is named like a private field to not be gut-wrenched across the codebase.

Comment (richvdh) [reply synapse/handlers/message.py]: oh, no you didn't. you just did the other changes in a commit called "Update copyright" which ... I didn't expect.

Comment (erikjohnston) [reply synapse/handlers/message.py]: Oh, blast, I think that might have been vscode doing the changes without saving the open files. Sigh.

Comment (erikjohnston) [reply synapse/handlers/message.py]: I have unfucked it now.

File: synapse/handlers/message.py (hunk 3 above)
Comment (erikjohnston) [reply synapse/handlers/message.py]: Its doable, but the diff would be quite big

**Pull request merged**